lib/tasks/inactive_providers.rake

```diff
@@ -0,0 +1,38 @@
+namespace :inactive_providers do
+  desc 'List inactive providers'
+  task list: :environment do
+    CSV.open(Rails.root.join('tmp/inactive_providers.csv'), 'wb') do |csv|
+      csv << %w[
+        id
+        email
+        last_sign_in_at
+        current_sign_in_at
+        created_at
+        updated_at
+        portal_enabled
+      ]
+      inactive_providers = Provider.where('current_sign_in_at < ? OR (current_sign_in_at IS NULL AND created_at < ?)', 6.months.ago.to_date, 6.months.ago.to_date)
```

Comment:
You could replace the ? with named variables to reduce duplication e.g. 
```ruby
Provider.where('current_sign_in_at < :limit OR (current_sign_in_at IS NULL AND created_at < :limit)', limit: 6.months.ago.to_date)
```

Reply:
Thanks @colinbruce have implemented this suggestion 👍